### PR TITLE
Improve modules promethues readme

### PR DESCRIPTION
**Pull request by HJianBo**
Fixes [EMQX-9574](https://emqx.atlassian.net/browse/EMQX-9574), [EMQX-9232](https://emqx.atlassian.net/browse/EMQX-9232).

### <samp>🤖 Generated by Copilot at 0abd5b9</samp>

This pull request adds a new README file for the `emqx_modules` application and updates the README file for the `emqx_prometheus` application. These changes aim to document and improve the usage and integration of these applications, which provide some optional features and metrics for the EMQ X broker. These changes are part of the refactoring and separation of these applications from the main `emqx` repo.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- ~Added tests for the changes~
- ~Changed lines covered in coverage report~
- ~Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files~
- [x] For internal contributor: there is a jira ticket to track this change
- ~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~
- ~Schema changes are backward compatible~

## Checklist for CI (.github/workflows) changes

- ~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- ~Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

File: apps/emqx_modules/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# EMQX Modules
+
+The application provide some minor functional modules that are not included in the MQTT
+protocol standard, including "Delayed Publish", "Topic Rewrite", "Topic Metrics" and "Telemetry".
+
+
+## Delayed Publish
+
+After enabling this module, messages sent by the user with the prefix
```
Comment (JimMoen) [inline apps/emqx_modules/README.md:9]: This module provides a feature that allows for manual intervention in the publishing delay of messages in an MQTT system.
After the module is enabled, messages sent by the MQTT clients with the topic

> nit: only mqtt clinets? how to delay a message for gateway clinets?

**Review by jinfahua — APPROVED** [review]
LGTM

File: apps/emqx_modules/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Topic Rewrite
+
+Topic Rewrite allows users to configure rules to change the topic strings that
+the client requests to subscribe or publish.
+
+This feature is very useful when designing topics that are compatible with different
+client versions. For example, an old device that has already been issued and cannot
+be upgraded may use old topic rules, but the production environment need to apply
+a new design rules for the topics.
```
Comment (JimMoen) [inline apps/emqx_modules/README.md:26]: This feature is useful to designing compatible topic system with different versions of client implements.
For example, a device will pub/sub message with topic `old/topic` with previous client system implement.
But the new devices implement doesn't contain the topic.
At this point, the topic-rewrite feature can be used to map the topics used by old devices to the topics used by the new system, achieving compatibility between new and old devices within the same system.

File: apps/emqx_modules/README.md (hunk 1 above)
Comment (HJianBo) [inline apps/emqx_modules/README.md:9]:
```suggestion
After enabling this module, messages sent by the clients with the topic prefixed with
```

File: apps/emqx_modules/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+See [Enabling/Disabling Delayed Publish via HTTP API](https://www.emqx.io/docs/en/v5.0/admin/api-docs.html#tag/MQTT/paths/~1mqtt~1delayed/put).
+
+
+## Topic Rewrite
+
+Topic Rewrite allows users to configure rules to change the topic strings that
+the client requests to subscribe or publish.
+
+This feature is very useful when designing topics that are compatible with different
+client versions. For example, an old device that has already been issued and cannot
```
Comment (HJianBo) [inline apps/emqx_modules/README.md:24]:
```suggestion
This feature is very useful when it need to compatibility with different versions of topic designs.
For example, an old device that has already been issued and cannot
```

File: apps/emqx_modules/README.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+be upgraded may use old topic rules, but the production environment need to apply
+a new design rules for the topics.
```
Comment (HJianBo) [inline apps/emqx_modules/README.md:26]:
```suggestion
be upgraded may use old topic rules, but for some reason, we adjusted the format of topics. We can use this feature to rewrite the old topics as the new formats to eliminate this difference.
```

**Review by JimMoen — APPROVED** [review]

File: apps/emqx_modules/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+See [Enabling/Disabling Delayed Publish via HTTP API](https://www.emqx.io/docs/en/v5.0/admin/api-docs.html#tag/MQTT/paths/~1mqtt~1delayed/put).
+
+
+## Topic Rewrite
+
+Topic Rewrite allows users to configure rules to change the topic strings that
+the client requests to subscribe or publish.
+
+This feature is very useful when it need to compatibility with different versions of topic designs.
```
Comment (thalesmg) [inline apps/emqx_modules/README.md:23]: Perhaps:

```suggestion
This feature is very useful when there's a need to transform between different topic structures.
```
?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_modules/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# EMQX Modules
+
+The application provide some minor functional modules that are not included in the MQTT
```
Comment (ieQu1) [inline apps/emqx_modules/README.md:3]:
```suggestion
The application provides some minor functional modules that are not included in the MQTT
```

**Review by thalesmg — APPROVED** [review]

**Pull request merged**